nomad/fsm.go

```diff
@@ -282,9 +282,9 @@ func (n *nomadFSM) Apply(log *raft.Log) interface{} {
 	case structs.ReconcileJobSummariesRequestType:
 		return n.applyReconcileSummaries(buf[1:], log.Index)
 	case structs.VaultAccessorRegisterRequestType:
-		return n.applyUpsertVaultAccessor(buf[1:], log.Index)
+		return nil
```

Comment:
We need to handle the `VaultAccessorSnapshot` restore code path in this file as well. We should do what we had to do recently for the timetable -- mark the snapshot enum as deprecated and make sure we're decoding the snapshot object but then throwing it away, so that we make progress thru the rest of the snapshot.

Reply:
Updated within: https://github.com/hashicorp/nomad/pull/25155/commits/1ee09ed6d31a4e94dcea6845809cc701e6765bb5